Enable Daikin energy_consumption sensors by deafult

declan1080 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

This resolves the issue reported on [https://github.com/home-assistant/core/issues/102451](url) by Enabling compressor_energy_consumption & heat_energy_consumption & compressor_frequency & compressor_energy_consumption. The default the code has specific values which set the mentioned entities to disabled when adding a Daikin AC unit I have updated this to enable by default.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

This PR fixes or closes issue: fixes #102451 
- This PR is related to issue: 
- Link to documentation pull request: 

Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Commit:
Enable Daikin energy_consumption sensors by deafult

Enabling compressor_energy_consumption & heat_energy_consumption & compressor_frequency & compressor_energy_consumption

frenck commented on homeassistant/components/daikin/sensor.py:
```diff
@@ -97,7 +97,7 @@ class DaikinSensorEntityDescription(SensorEntityDescription, DaikinRequiredKeysM
         device_class=SensorDeviceClass.ENERGY,
         native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
-        entity_registry_enabled_default=False,
+        entity_registry_enabled_default=True,
```
These are the defaults, so can be removed instead.

```suggestion
```

Commit:
Update homeassistant/components/daikin/sensor.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Commit:
Remove entity_registry_enabled_default=True/False

Remove "entity_registry_enabled_default=True/False" from the following entites:

compressor_energy_consumption & heat_energy_consumption & compressor_frequency & compressor_energy_consumption

declan1080 reviewed (commented):
Removed the defaults @frenck

Commit:
Merge branch 'dev' into patch-1

disforw commented:
@declan1080 can you please explain why these entities should be enabled by default?
I think the idea behind disabling them is that most people who add this integration only care about control of their unit, not necessarily monitoring its power consumption. It is relatively easy to enable the entities after adding for the people that want it.

declan1080 commented:
Hey there, I agree with you that most people are using this to just control their units but those who do have the energy system setup etc are potentially missing out on this data in the first instance and they would already have to know the entities are disabled by default.

Maybe this can be tackled with a message/option to appear during the setup process?

disforw commented:
Keep in mind not everyones units report energy usage (mine do not), this is explained in the documentation. Adding a prompt at setup may be confusing for the people who get zero values.
